contrib/storage-hbase/src/main/java/org/apache/drill/exec/store/hbase/HBaseRecordReader.java

```diff
@@ -84,12 +84,16 @@ public class HBaseRecordReader extends AbstractRecordReader implements DrillHBas
 
   private final Connection connection;
 
-  public HBaseRecordReader(Connection connection, HBaseSubScan.HBaseSubScanSpec subScanSpec, List<SchemaPath> projectedColumns) {
+  public HBaseRecordReader(Connection connection, HBaseSubScan.HBaseSubScanSpec subScanSpec, List<SchemaPath> projectedColumns, int maxRecords) {
     this.connection = connection;
     hbaseTableName = TableName.valueOf(
         Preconditions.checkNotNull(subScanSpec, "HBase reader needs a sub-scan spec").getTableName());
     hbaseScan = new Scan(subScanSpec.getStartRow(), subScanSpec.getStopRow());
     hbaseScanColumnsOnly = new Scan();
+    //todo Whether to remove if condition judgment
```

Comment:
I'm not sure what this comment means but please remove or explain further.   Just so you are aware, Calcite does provide the logic as to when to push the limit down and when not to so I think we are good here.  To verify this, you can add a unit test with the query:

```sql
SELECT * 
FROM hbase
LIMIT 10 OFFSET 5
```
and you should see the `maxRecords` parameter set to 15. 

If you look at the unit tests for the Phoenix plugin, you can see a good example of this:

https://github.com/apache/drill/blob/78a52087bb85bcc7034157544193155b87503111/contrib/storage-phoenix/src/test/java/org/apache/drill/exec/store/phoenix/PhoenixSQLTest.java#L92-L128